packages/perspective-viewer-d3fc/src/js/legend/styling/enableDragging.js

```diff
@@ -11,28 +11,61 @@ import * as d3 from "d3";
 import {isElementOverflowing} from "../../utils/utils";
 import {getChartElement} from "../../plugin/root";
 
+const margin = 10;
+const pinned = "pinned";
+const resizeForDraggingEvent = "resize.for-dragging";
+
 export function enableDragging(element) {
     const node = element.node();
     node.style.cursor = "move";
 
     const drag = d3.drag().on("drag", function() {
-        overrideDefaultResizingBehaviour(node);
         const [offsetX, offsetY] = enforceContainerBoundaries(this, d3.event.dx, d3.event.dy);
         this.style.left = `${this.offsetLeft + offsetX}px`;
         this.style.top = `${this.offsetTop + offsetY}px`;
+
+        const element = d3.select(this);
+        if (isNodeInTopRight(node)) {
+            pinNodeToTopRight(node, element);
+            return;
+        }
+
+        unpinNodeFromTopRight(node, element);
```

Comment:
I still don't think we should be re-registering the "resize" event handler every time we get a "drag" event. This may work, but it just feels wrong to me :-(

Reply:
There's a switch on the "pinned" attr preventing this.

Reply:
Where is the switch?

Reply:
I don't see why @DevAndyLee's suggestion wouldn't work

Reply:
You can ignore my comments, there's some new code that hasn't been pushed.